cli/azd/internal/appdetect/java_test.go

```diff
@@ -0,0 +1,154 @@
+package appdetect
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestJavaDetector_Language(t *testing.T) {
+	jd := &javaDetector{}
+	if jd.Language() != Java {
+		t.Errorf("expected language to be Java, got %v", jd.Language())
+	}
+}
+
+func TestJavaDetector_DetectProject_WithPomXml(t *testing.T) {
+	jd := &javaDetector{}
+	entries := []fs.DirEntry{
+		mockDirEntry{name: "pom.xml"},
+	}
+	tempDir := t.TempDir()
+	err := os.WriteFile(filepath.Join(tempDir, "pom.xml"), []byte(`
+		<project>
+			
+		</project>`), 0600)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	project, err := jd.DetectProject(context.Background(), tempDir, entries)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	if project == nil {
+		t.Fatal("expected project to be detected, got nil")
+	}
+}
+
+func TestJavaDetector_DetectProject_WithoutPomXml(t *testing.T) {
```

Comment:
If it make sense to you: this code path is already covered by the acceptance tests in `appdetect_test.go` by the fact that we'd call a similar codepath on each sub directory.